### PR TITLE
Fix deletion of collection types when they are orphaned

### PR DESCRIPTION
**Pull request by msullivan**
This fixes a bunch of test_schema tests with DDL migrations turned on.

### CODE REVIEW

File: edb/schema/delta.py [hunk 1]
```diff
@@ -2715,6 +2715,11 @@ class DeleteObject(ObjectCommand[so.Object_T], Generic[so.Object_T]):
     #: in the schema.
     if_unused = struct.Field(bool, default=False)
 
+    #: Potential references to this object that we know are being
+    #: deleted, which we use to resolve if_unused.
+    expiring_refs = struct.Field(AbstractSet[so.Object],
+                                 default=frozenset())
```
Comment (1st1) [inline edb/schema/delta.py:2721]: Should this be an `ephemeral` field?

Comment (elprans) [reply edb/schema/delta.py:2721]: It's not a `schema.Field` on an object, just a delta command field.

File: edb/schema/types.py [hunk 2]
```diff
@@ -861,6 +865,7 @@ def get_schema_class_displayname(cls) -> str:
     def as_colltype_delete_delta(
         self,
         schema: s_schema.Schema,
+        expiring_refs: AbstractSet[so.Object],
```
Comment (elprans) [inline edb/schema/types.py:868]: Let's make this a kwarg

**Comment by msullivan** [comment]
Any more feedback, or is this ready to merge?

**Review by elprans — APPROVED** [review]
LGTM

**Pull request merged**